spec/volcanic/authenticator/v1/scope_spec.rb

```diff
@@ -0,0 +1,211 @@
+# frozen_string_literal: true
+
+RSpec.describe Volcanic::Authenticator::V1::Scope do
+  let(:scope) { 'vrn:local:-1:identity/1?user_id=123' }
+
+  subject { described_class.parse(scope) }
+
+  describe 'method' do
+    describe '#parse' do
+      it 'returns correctly' do
+        expect(subject.class).to eq(described_class)
+      end
+    end
+
+    describe '#initialize' do
+      it 'assigns correctly' do
+        expect(subject.to_s).to eq(scope)
+        expect(subject.stack_id).to eq('local')
+        expect(subject.dataset_id).to eq('-1')
+        expect(subject.resource).to eq('identity')
+        expect(subject.resource_id).to eq('1')
+        expect(subject.qualifiers).to eq('user_id=123')
+      end
+
+      context 'missing elements have defaults' do
+        let(:scope) { 'vrn:*:*:*' }
+
+        it 'assigned correctly' do
+          expect(subject.to_s).to eq('vrn:*:*:*')
+          expect(subject.stack_id).to eq('*')
+          expect(subject.dataset_id).to eq('*')
+          expect(subject.resource).to eq('*')
+          expect(subject.resource_id).to be_nil
+          expect(subject.qualifiers).to be_nil
+        end
+      end
+    end
+
+    describe '.include?(other)' do
+      subject { described_class.parse(scope).include?(other) }
+
+      context 'with an exact match' do
+        let(:scope) { 'vrn:local:-1:jobs/5' }
+
+        context 'when other: vrn:local:-1:jobs/*' do
+          let(:other) { 'vrn:local:-1:jobs/*' }
+
+          it 'returns correctly' do
```

Comment:
this is a good usecase for one-liner syntax... the 'returns correctly' doesn't add anything (and, if anything, distracts because the thing I actually need to know is that it returns false - "correctly" is meaningless)

`it { is_expected.to eq(false) }`